pkg/controller/gittrack/gittrack_controller.go

```diff
@@ -134,12 +146,14 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	err = c.Watch(&source.Kind{Type: &farosv1alpha1.ClusterGitTrackObject{}}, &handler.EnqueueRequestForOwner{
-		IsController: true,
-		OwnerType:    &farosv1alpha1.ClusterGitTrack{},
-	})
-	if err != nil {
-		return err
+	if opts.clusterGitTrackMode != farosflags.CGTMDisabled {
+		err = c.Watch(&source.Kind{Type: &farosv1alpha1.ClusterGitTrackObject{}}, &handler.EnqueueRequestForOwner{
```

Comment:
Shouldn't there also be a watch for `GitTrackObjecsts` owned by `ClusterGitTracks` which will also need gating?